### PR TITLE
Matt base files

### PR DESCRIPTION
**Pull request by mattagape**
Merge my changes to `generator/base_files` and `generator/cmake_files`

### CODE REVIEW

File: generator/base_files/BaseFile.py [hunk 1]
```diff
@@ -110,16 +124,24 @@ def create_lines(self, line, tabsize, is_comment=False):
...
-        elif num_words == 1:
+        if num_words < 2:
             lines.append(line)
         else:
             self.parse_lines(lines, words, max_length)
         return lines
 
     @staticmethod
     def parse_lines(lines, words, max_length):
+        """
```
Comment (skeating) [inline generator/base_files/BaseFile.py:135]: Bane of my life !!!!!

File: generator/base_files/BaseTexFile.py [hunk 2]
```diff
@@ -97,45 +109,84 @@ def sort_class_names(self, classes):
...
+
+        :param my_list: the list which we update.
+        """
+        for j in range(0, len(my_list)):
+            if my_list[j]['type'] in ['lo_element', 'element',
+                                      'inline_lo_element']:
+                name = my_list[j]['element']
+            else:
+                name = my_list[j]['name']
+            texname = strFunctions.texify(name)
```
Comment (skeating) [inline generator/base_files/BaseTexFile.py:157]: not SF ? :-)

**Review by skeating — APPROVED** [review]
Really like the refactoring - very nice.
One comment about if we are going to import strfunctions as SF should probably do that everywhere; but ahead and merge.
I'll get to the TODOs eventually

File: generator/base_files/BaseTexFile.py (hunk 2 above)
Comment (mattagape) [reply generator/base_files/BaseTexFile.py:157]: I suppose I should have done, but IIRC it was only used once or twice in this module, so I didn't bother.

**Pull request merged**